src/Auth/Actions/userActions.js

```diff
@@ -76,40 +69,45 @@ export const LoginUser = (credentials, history, setFieldError, setSubmitting) =>
                 const token = userData.data.token
                 localStorage.setItem('token', token)
                 history.push("/dashboard")
-                dispatch(loginUserSuccess(userData))
+                return dispatch => {
+                    dispatch(loginUserSuccess(data))
+                }
             }
             setSubmitting(false);
-        })
-        .catch(err => console.error(err));
+    } catch (error) {
+        throw new Error(error)
     }
 }
 
-export const RegisterUser = (credentials, history, setFieldError, 
-    setSubmitting)  => {
-        
-        return () => {
-        axios.post(`${SERVER}auth/v1/create-user`,
-        credentials,
-        {
-            headers: {
-                "Content-Type": "application/json"
-            }
+
+export const  getUserDetails = async () => {
```

Comment:
rewrite like this. notice where the dispatch is. the way you are writing you have to then declare dispatch twice. I think it was my fault am sorry for that

`export function getUser() {
    return dispatch => {
        const promise = getApi('auth/profile');
        dispatch(getUserRequest());
        promise.then(
            function (payload) {
                dispatch(getUserSuccess(payload.data));
            },
            function (error) {
                dispatch(getUserError(error));
            }
        );

        return promise;
    };
}`

Reply:
Ok sir